src/qat/model/device.py

```diff
@@ -1,118 +1,272 @@
 from __future__ import annotations
 
+import uuid
 from typing import Optional
 
-from pydantic import Field
+import numpy as np
+from pydantic import Field, model_validator
+from pydantic_core import core_schema
 
-from qat.model.component import Component
-from qat.model.serialisation import Ref, RefDict, RefList
-from qat.purr.compiler.devices import ChannelType
+from qat.utils.pydantic import WarnOnExtraFieldsModel
+
+QubitId = int
+
+
+class CalibratablePositiveFloat(float):
```

Comment:
there are two important ideas just from the name:

1. Tunability of a float number
2. The float remaining positive during the tuning process

Could explain why do you need the second point ?

Reply:
This variable is used for frequencies and sample times, which are by definition never lower than zero (unless I'm horribly mistaken?). So we validate against the positivity condition